cpp/bench/ann/CMakeLists.txt

```diff
@@ -86,6 +86,14 @@ endif()
 # ##################################################################################################
 # * Configure tests function-------------------------------------------------------------
 
+# Build and link static version of the GBench to keep ANN_BENCH self-contained.
```

Comment:
This won't work when google benchmark is pre-built and coming from the environment. The sources in that case will be empty and you will fail to have any gtest/gbench in your final executable.